internal/app/admin/http/key.go

```diff
@@ -0,0 +1,44 @@
+package handler
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/envoyproxy/xds-relay/internal/app/orchestrator"
+	"github.com/envoyproxy/xds-relay/internal/pkg/util/stringify"
+	"github.com/envoyproxy/xds-relay/pkg/marshallable"
+)
+
+func keyDumpHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		allKeys, err := orchestrator.Orchestrator.GetDownstreamAggregatedKeys(*o)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			errMessage, _ := stringify.InterfaceToString(&marshallable.Error{
+				Message: fmt.Sprintf("error in getting cache keys: %s", err.Error()),
+			})
+			_, _ = w.Write([]byte(errMessage))
```

Comment:
Similar comment with err handling as the EDS one.

Reply:
here the header is already 500, there would be no difference in making the check